rero_ils/modules/items/api/record.py

```diff
@@ -73,6 +77,17 @@ def extended_validation(self, **kwargs):
             - temporary_item_type isn't in the future (if specified)
 
         """
+        from . import ItemsSearch
+        if barcode := self.get('barcode'):
+            filters = Q('term', barcode=barcode)
+            results = ItemsSearch()\
+                .filter('bool', must=[filters])\
+                .source(['pid'])\
+                .scan()
```

Comment:
It isn't necessary to create `Q` in this case. 
If you need only one field, you can avoid brackets

You could simply write
```python
results = ItemsSearch()\
    .filter('term', barcode=barcode)\
    .source('pid').scan()
```